pkg/engine/request_performance_tests/play_game.go

```diff
@@ -0,0 +1,75 @@
+package requestperformancetests
+
+import (
+	"testing"
+
+	"github.com/YetAnotherSpieskowcy/Carcassonne-Engine/pkg/engine"
+	"github.com/YetAnotherSpieskowcy/Carcassonne-Engine/pkg/tilesets"
+)
+
+type TestedRequest func(games *[]engine.SerializedGameWithID, eng *engine.GameEngine, b *testing.B)
+
+func PlayGame(gameCount int, b *testing.B, testedRequest TestedRequest) {
+	b.StopTimer()
+
+	eng, err := engine.StartGameEngine(4, "")
+	if err != nil {
+		b.Fatal()
+	}
+
+	var games = []engine.SerializedGameWithID{}
+	for seed := range gameCount {
+		game, err := eng.GenerateSeededGame(tilesets.StandardTileSet(), int64(seed+1000))
+		if err != nil {
+			b.Fatal()
+		}
+		games = append(games, game)
+	}
+
+	// for each turn
+	for range len(tilesets.StandardTileSet().Tiles) {
+
+		// at start before makiny any turn
+		// test desired requests
+		for range b.N {
+			testedRequest(&games, eng, b)
+		}
```

Comment:
Asking just to make sure - this is here so that we can test the requests in a "real" environment, during an actual game, instead of on an engine that has just started, right?

(At first I wasn't sure why do we need the whole PlayGame method, when only this line is the one that actually measures anything)

Reply:
We want to measure the average time of a certain request during whole game, so we need to play it, and measure the request each turn. We test multiple different games at once, to have more reliable measurements. Moreover we check all turns, instead of single case, witch would some previous special setup.